### ISSUE TITLE
Finalize Filters

### ISSUE THREAD

**Issue by Jose-S** [sandboxnu/pharmd-tracker#29]
**Problem**: Finish cleaning and documenting the filter component code.
**Solution**: Add types and ensure functionality works.

**Description**: Filters allow the deans to filter students based on their properties. These filters can be combined and provide a powerful way too easily query the database. Filters are currently functioning, but need to be properly documented and finalized.

**Specs:**
- [ ] Ensure filters work
- [ ] Document components
- [ ] Break up components
- [ ] Add typescript

**Notes**
Some components are made up of smaller components, so make sure to separate components into separate files. For example, if a filter uses a multi-select dropdown, then there should be a multi-select dropdown component that can be used independently of the filter. 

If blocked, contact @Jose-S or @kev1n80

**Comment by Jose-S** [comment]
Hey, this issue is about the work you mentioned you were doing yesterday.

**Comment by kev1n80** [comment]
Yup, sounds great! I was wondering if you had a guideline for the
documentation.

**Comment by Jose-S** [comment]
Here is a [gist](https://gist.github.com/Jose-S/95c0d255d37baff8a4387946fbe013ce) showing how a well formatted and documented component should look like. Let me know if you think we should add something else.

**Comment by kev1n80** [comment]
Looks great, thank you!
